### PR TITLE
M5843 Import package selection

### PR DESCRIPTION
**Pull request by dennishendriksen**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- (n.a.) User documentation updated
- (n.a.) (If you have changed REST API interface) view-swagger.ftl updated
- (n.a.) Test plan template updated
- [x] Clean commits (squash)
- [x] Added Feature/Fix to release notes

### CODE REVIEW

**Comment by dennishendriksen** [comment]
test this please

**Comment by dennishendriksen** [comment]
test this please

**Comment by dennishendriksen** [comment]
test this please

**Comment by dennishendriksen** [comment]
test this please

**Comment by tommydeboer** [comment]
1. Add a group "Test"
2. Add user and give permission to: 
    - VIEW to Data Explorer, Importer
    - WRITE to importRun, Attribute, EntityType, Package
3. Maker user group manager
4. Login as user
5. Try to import it_emx_autoid

Error message:
Only superusers are allowed to create packages without a parent.

While technically correct, this is not the most useful feedback. Can we change it to something like:

"You can only upload data to packages of groups you are a member of."

**Comment by tommydeboer** [comment]
1. Add group "Test"
2. Add user and give permission to:
    - VIEW to Data Explorer, Importer
    - WRITE to importRun, Attribute, EntityType, Package
3. Maker user group MANAGER 
4. Login as user
5. Try to import autoid.xlsx
===> Can't select target package in Packages step

File: molgenis-api-tests/src/test/java/org/molgenis/api/tests/utils/RestTestUtils.java [hunk 1]
```diff
@@ -144,7 +157,7 @@ private static String uploadEMXFile(String adminToken, File file)
 	{
 		String importJobStatusUrl = given().multiPart(file)
 										   .param("file")
-										   .param("action", "ADD_UPDATE_EXISTING")
+										   .param("action", "ADD_UPDATE_EXISTING").param("packageId", "base")
```
[the remark this answers is not in the document]
Comment (dennishendriksen) [reply molgenis-api-tests/src/test/java/org/molgenis/api/tests/utils/RestTestUtils.java]: Strange, reformat-on-commit is enabled in my IntelliJ. Anyways: fixed

**Comment by dennishendriksen** [comment]
> ===> Can't select target package in Packages step

The user needs to have write permissions on all entity types in package `sys_md`.

**Comment by dennishendriksen** [comment]
> Can we change it to something like:
> "You can only upload data to packages of groups you are a member of."

You can upload data to packages outside of your group if you have the right permissions. I wonder why the exception thrown by `PackageRepositorySecurityDecorator` is not a regular `PackagePermissionDeniedException` which would be clear to the user. We should either replace it or change the exception message of `NullParentPackageNotSuException`. I propose to do that separately from this feature.

**Comment by tommydeboer** [comment]
Tommy:
>Add group "Test"
Add user and give permission to:
VIEW to Data Explorer, Importer
WRITE to importRun, Attribute, EntityType, Package
Maker user group MANAGER
Login as user
Try to import autoid.xlsx
===> Can't select target package in Packages step

Dennis:
>The user needs to have write permissions on all entity types in package sys_md.

The problem seems to be that the group is called 'test' and that the entity is called 'testAutoId'. At least I think so because when I use a group "My Group" I can use the feature as expected.

**Comment by dennishendriksen** [comment]
Default build failed due to:
```
[INFO] $ cross-env BABEL_ENV=test karma start test/unit/karma.conf.js --single-run
[ERROR] Hash: 5938d583407b6824883a
[ERROR] Version: webpack 3.11.0
[ERROR] Time: 278629ms
[ERROR]    Asset     Size  Chunks                    Chunk Names
[ERROR] index.js  5.46 MB       0  [emitted]  [big]  index.js
[ERROR]   [14] (webpack)/buildin/global.js 509 bytes {0} [built]
[ERROR]   [33] ./node_modules/vue/dist/vue.esm.js 292 kB {0} [built]
[ERROR]   [41] ./src/components/AppCard.vue 2 kB {0} [built]
[ERROR]   [68] ./src/pages/AppManagerContainer.vue 2.04 kB {0} [built]
[ERROR]   [69] ./src/utils/filter-array.js 4.4 kB {0} [built]
[ERROR]   [76] ./src/store/actions.js 10.8 kB {0} [built]
[ERROR]   [81] ./src/store/mutations.js 2.01 kB {0} [built]
[ERROR]  [111] ./src/components/AppCardGallery.vue 1.66 kB {0} [built]
[ERROR]  [194] ./src/AppManager.vue 1.62 kB {0} [optional] [built]
[ERROR]  [197] ./test/unit/index.js 657 bytes {0} [built]
[ERROR]  [198] ./test/unit/specs \.spec$ 309 bytes {0} [built]
[ERROR]  [199] ./test/unit/specs/components/AppCard.spec.js 2.12 kB {0} [optional] [built]
[ERROR]  [423] ./test/unit/specs/pages/AppManagerContainer.spec.js 2.79 kB {0} [optional] [built]
[ERROR]  [492] ./test/unit/specs/store/mutations.spec.js 1.14 kB {0} [optional] [built]
[ERROR]  [494] ./src ^\.\/(?!main(\.js)?$) 632 bytes {0} [built]
[ERROR]     + 481 hidden modules
[ERROR] 
[ERROR] WARNING in ./node_modules/testdouble/lib/replace/module.js
[ERROR] 29:15-54 Critical dependency: the request of a dependency is an expression
[ERROR]  @ ./node_modules/testdouble/lib/replace/module.js
[ERROR]  @ ./node_modules/testdouble/lib/replace/index.js
[ERROR]  @ ./node_modules/testdouble/lib/index.js
[ERROR]  @ ./test/unit/specs/components/AppCard.spec.js
[ERROR]  @ ./test/unit/specs \.spec$
[ERROR]  @ ./test/unit/index.js
[ERROR] 
[ERROR] WARNING in ./node_modules/testdouble/lib/replace/module.js
[ERROR] 33:15-35:17 Critical dependency: the request of a dependency is an expression
[ERROR]  @ ./node_modules/testdouble/lib/replace/module.js
[ERROR]  @ ./node_modules/testdouble/lib/replace/index.js
[ERROR]  @ ./node_modules/testdouble/lib/index.js
[ERROR]  @ ./test/unit/specs/components/AppCard.spec.js
[ERROR]  @ ./test/unit/specs \.spec$
[ERROR]  @ ./test/unit/index.js
```

**Comment by tommydeboer** [comment]
For future reference:

Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, Attribute, EntityType, Package
Login as user
Try to import it_emx_autoid
===> no writable packages 


Add group "Test"
Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, Attribute, EntityType, Package
Maker user group MANAGER
Login as user
Try to import it_emx_autoid
===> Only superusers are allowed to create packages without a parent.


Add group "Test"
Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, Attribute, EntityType, Package
Maker user group MANAGER
Login as user
Try to import test_autoid
===> Success!


Add group "Test"
Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, Attribute, EntityType, Package
Maker user group EDITOR 
Login as user
Try to import test_autoid
===> No packages exist that allow for adding entity types and packages.


Add group "Test"
Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, WRITEMETA to Attribute, EntityType, Package
Maker user group MANAGER 
Login as user
Try to import autoid.xlsx
===> Can't select target package in Packages step (FIXED)


Add group "Test"
Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, WRITEMETA to sys_md
Maker user group MANAGER 
Login as user
Try to import variantsWithSamples.vcf
===> Possibility to select target package (test / test_autoid) + import successful


Add group "My Group"
Add user and give permission to:
  VIEW to Data Explorer, Importer
  WRITE to importRun, WRITEMETA to sys_md
Maker user group MANAGER 
Login as user
Try to import autoid.xlsx
===> Possibility to select target package (test / test_autoid) + import successful

**Pull request merged**